homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -0,0 +1,96 @@
+"""Plugwise Binary Sensor component for Home Assistant."""
+
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import callback
+
+from .const import DOMAIN, FLAME_ICON, FLOW_OFF_ICON, FLOW_ON_ICON, IDLE_ICON
+from .sensor import SmileSensor
+
+BINARY_SENSOR_MAP = {
+    "dhw_state": ["Domestic Hot Water State", None],
+    "slave_boiler_state": ["Secondary Heater Device State", None],
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smile binary_sensors from a config entry."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    entities = []
+
+    all_devices = api.get_all_devices()
+    for dev_id, device_properties in all_devices.items():
+        if device_properties["class"] == "heater_central":
+            data = api.get_device_data(dev_id)
+            for binary_sensor, dummy in BINARY_SENSOR_MAP.items():
+                if binary_sensor in data:
```

Comment:
We would benefit from using guard clauses here to save on indentation. Ie invert the checks and continue on true. Then outdent below.

Reply:
Drafted #36406 will continue (and test) on it later today